### DISCUSSION TITLE
timezone in browser plugins

### DISCUSSION THREAD

**Discussion by MehrCurry** [dani-garcia/vaultwarden#4230] [Q&A]
I have configured 2FA and i am able to login with the web application and the mac application. But when trying to login with the brower plugins (firefox, safari, brave) i get an error. The error message contains the utc time on the server. Timezone is set and diagnostics looks good for me. Any hints what might be wrong?
<img width="935" alt="grafik" src="https://github.com/dani-garcia/vaultwarden/assets/328753/fbbb6e7c-e4d4-4042-b8bb-52921b74e1ed">
<img width="380" alt="grafik" src="https://github.com/dani-garcia/vaultwarden/assets/328753/47c2a7b1-aac6-4a7e-adb3-cf0e15340414">

**Comment by Node815** [comment]
I had to switch my 2FA to send Email instead.  For some reason, it's using the UTC timezone only for the CLI login option.  When it sent via email, it accepted it since the server generated it instead.   

I did log into the container and confirm using the ```date``` command that the timezone is correct and matched my local time.  So, it may or may not be a bug in the software.

**Reply by BlackDex** [reply]
It's not a bug in the software.
Somewhere your date time is of. Check the Vaultwarden Admin Interface `/admin/diagnostics` voor more details.

**Reply by Node815** [reply]
I already did that and came here.  It matched my local time zone and the UTC Time both ways.  
![image](https://github.com/user-attachments/assets/30d32f08-018d-4760-a5ee-7119f8e94b5b)

My local time is -7/-8 hours from UTC.    Depending on DST.  Right now, it's -7.

**Reply by BlackDex** [reply]
Ok. And where did the 2FA TOTP code cube from? Your mobile device? Is that still in sync? Check via https://time.is/ on that device if it is still correct.

**Reply by Node815** [reply]
Mobile device (Android) using "ente auth". 

[
![share_1164522476789469612](https://github.com/user-attachments/assets/30eddae7-0e45-4696-a418-5feaf77fd32c)


And my Desktop where I tried to login from:
![image](https://github.com/user-attachments/assets/aab4d34d-dec4-4ef3-958f-581cbfdd6214)

**Reply by BlackDex** [reply]
Looks all ok. All in a range that shouldn't matter. 
Then i have no idea why that is. Maybe a small hiccup.
